andela/Ah-backend-xmen
# #162725275 Comments should track edit history

#### What does this PR do?
This PR enables users to be able to see the previous versions of comments made on an article after previous edits.

#### Description of Task to be completed?
The previous version of a modified comment should be retrievable on demand by either the author of the article or the author of the comment.


#### How should this be manually tested?
- First things first, fetch this branch by running the command `git fetch origin ft-comments-track-edit-history-162725275` in the terminal.
- Check into this branch by running `git checkout ft-comments-track-edit-history-162725275`
- Activate your virtual environment by running `source venv/bin/activate`
- Install the django-simple-history dependancy by running `pip install -r requirements.txt`
- Apply the new changes to the models by running `python manage.py migrate`
- Start the application by running `python manage.py runserver`
- Click the `Run in postman` button to test the endpoint as shown in the screenshots below

The PR addresses `GET/api/articles/<slug>/comments/<comment_id>/history` endpoint

#### What are the relevant pivotal tracker stories?
[#162725275](https://www.pivotaltracker.com/story/show/162725275)

#### Screenshots

#### Comment edit history
![image](https://user-images.githubusercontent.com/36506567/51970693-9d04cb00-2488-11e9-91ab-0bcb9cc3978b.png)

#### Reply edit history
![image](https://user-images.githubusercontent.com/36506567/51965655-46dd5b00-247b-11e9-9be4-46bab42d5f07.png)

#### POSTMAN
[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/f24ca45d27c7253a80d0)
